go/teams/seitan_test.go

```diff
@@ -113,10 +140,22 @@ func TestSeitanKnownSamples(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, expectedInviteID, inviteID)
 
-	peiKey2, _, err := ikey.generatePackedEncryptedIKeyWithSecretKey(secretKey, keybase1.PerTeamKeyGeneration(0), peiKey.RandomNonce)
+	label := ikeyAndLabelV1.L
+	labelType, err := label.T()
 	require.NoError(t, err)
-	require.Equal(t, peiKey.Version, peiKey2.Version)
-	require.Equal(t, peiKey.TeamKeyGeneration, peiKey2.TeamKeyGeneration)
-	require.Equal(t, peiKey.RandomNonce, peiKey2.RandomNonce)
-	require.Equal(t, peiKey.EIKey, peiKey2.EIKey)
+	require.Equal(t, keybase1.SeitanIKeyLabelType_SMS, labelType)
+
+	labelSms := label.Sms()
+	require.Equal(t, "Edwin Powell Hubble", labelSms.F)
+	require.Equal(t, "+48123ZZ3045", labelSms.N)
+
+	// Packing struct is non-deterministic as far as field ordering is
+	// concerned, so we will not be able to get same ciphertext here.
+
+	// peiKey2, _, err := ikey.generatePackedEncryptedIKeyWithSecretKey(secretKey, keybase1.PerTeamKeyGeneration(1), peiKey.RandomNonce, ikeyAndLabelV1.L)
+	// require.NoError(t, err)
+	// require.Equal(t, peiKey.Version, peiKey2.Version)
+	// require.Equal(t, peiKey.TeamKeyGeneration, peiKey2.TeamKeyGeneration)
+	// require.Equal(t, peiKey.RandomNonce, peiKey2.RandomNonce)
+	// require.Equal(t, peiKey.EncryptedIKeyAndLabel, peiKey2.EncryptedIKeyAndLabel)
```

Comment:
Just wanted to confirm/discuss my findings and I will remove commented code.

Reply:
It is deterministic, based on the order of keys, isn't it?

Reply:
You're right. Somehow I thought it sorts keys alphabetically and I was surprised to see different payloads after packing. I managed to create exactly same payload using iced code by rearranging label struct to match go's variant. I also realized that we serialize IKey as string on go side but on the iced side it's passed as `Buffer`. Should we use `[]byte` in go as well?

Reply:
I'd say switch iced to use `string`, that feels like the right answer, no?

Reply:
I think so, yes. Thanks